objects/src/notes/metadata.rs

```diff
@@ -19,12 +19,39 @@ pub struct NoteMetadata {
     sender: AccountId,
     tag: Felt,
     note_type: NoteType,
+    aux: Felt,
 }
 
 impl NoteMetadata {
     /// Returns a new [NoteMetadata] instantiated with the specified parameters.
-    pub fn new(sender: AccountId, note_type: NoteType, tag: Felt) -> Self {
-        Self { sender, tag, note_type }
+    pub fn new(
+        sender: AccountId,
+        note_type: NoteType,
+        tag: Felt,
+        aux: Felt,
+    ) -> Result<Self, NoteError> {
+        match note_type {
+            NoteType::OffChain => {
+                // 1st most significant bit set to 0
+                if tag.as_int() >> 63 != 0 {
+                    return Err(NoteError::InvalidOffchainTag);
+                }
+            },
+            NoteType::Public => {
+                // 2nd most significant bit set to 0
+                if (tag.as_int() >> 62) & 1 != 0 {
+                    return Err(NoteError::InvalidPublicTag);
+                }
+            },
+            NoteType::Encrypted => {
+                // 1st & 2nd most significant bit set to 0
+                if tag.as_int() >> 62 != 0 {
```

Comment:
Incorrect validation. This will return `true` for `0b01` and `0b10` as well. I would change this line to:
```rust
if tag.as_int() >> 62 == 0b11 {
```

Reply:
> This will return true for 0b01 and 0b10 as well

Humm, I think this is correct, no? The two MSb should be set to `0`, so the combinations `0b11`, `0b10`, and `0b01` should be true here.

Reply:
@hackaugusto , oh, my bad, it is correct!